node/src/types/block/past_finality_signatures.rs

```diff
@@ -0,0 +1,268 @@
+use std::collections::BTreeSet;
+
+use casper_types::{
+    bytesrepr::{self, Bytes, FromBytes, ToBytes},
+    PublicKey,
+};
+
+use datasize::DataSize;
+use itertools::Itertools;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use tracing::error;
+
+/// List of identifiers for finality signatures for a particular past block.
+///
+/// That past block height is current_height - rewards_lag, the latter being defined
+/// in the chainspec.
+///
+/// We need to wait for a few blocks to pass (`rewards_lag`) to store the finality
+/// signers because we need a bit of time to get the block finality.
+#[derive(
+    Clone,
+    DataSize,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    Hash,
+    Serialize,
+    Deserialize,
+    JsonSchema,
+    Debug,
+    Default,
+)]
+pub struct PastFinalitySignatures(Vec<u8>);
```

Comment:
As discussed in standup, I would like for us to workshop better terminology. We can merge with this name for now, but by the time we ship it we will ideally have refactored it to a less vague name.

That aside, I think the payload of this tuple struct should be a map keyed by block hash or block height and a value of Vec<u8>. This will allow the discussed scheme of permitting proposers to include signatures from earlier blocks which were missed by the proposer of that block.